### PR TITLE
Clarify missing RID error

### PR DESCRIPTION
**Pull request by JanKrivanek**
### Context

Current error might get confusing in some cases. Especially when users are trying to st value of `RuntimeIdentifiers` conditionally based on configurations.

### Changes made

Just adjusting the error message - stealing the finding and formulation from @rainersigwald  - explicitly mentioning the `RuntimeIdentifiers` is expected to contain all supported RIDs

### CODE REVIEW

File: src/Microsoft.NuGet.Build.Tasks/Strings.resx [hunk 1]
```diff
@@ -147,7 +147,7 @@
     <comment>0 package id</comment>
   </data>
   <data name="MissingRuntimeIdentifierInProjectFile" xml:space="preserve">
-    <value>Your project file doesn't list '{0}' as a "RuntimeIdentifier". You should add '{1}' to the "RuntimeIdentifiers" property in your project file and then re-run NuGet restore.</value>
+    <value>Your project file doesn't list '{0}' as a "RuntimeIdentifier". You should add '{1}' to the "RuntimeIdentifiers" property in your project file (RuntimeIdentifiers should always be the complete list of RIDs supported by the project, regardless of the current configuration, platform or target. Single unconditioned RuntimeIdentifiers property is recommended) and then re-run NuGet restore.</value>
```
Comment (zivkan) [inline src/Microsoft.NuGet.Build.Tasks/Strings.resx:150]: This feels very wordy, and the information in the parenthesis, which are two sentences, is in the middle of the "outer" sentence. I think that might make it a bit more difficult to read. You could put everything in parenthesis as new sentences after the existing message. It's still a fair amount of text, but maybe that's ok.

```suggestion
    <value>Your project file doesn't list '{0}' as a "RuntimeIdentifier". You should add '{1}' to the "RuntimeIdentifiers" property in your project file and then re-run NuGet restore. RuntimeIdentifiers should always be the complete list of RIDs supported by the project, regardless of the current configuration, platform or target. A single unconditioned RuntimeIdentifiers property is recommended</value>
```

Does this error have a unique MSB code with a docs page? I don't know what the balance should be between "lots of details in the error message" vs "give a concise message, and expect customers to find the docs page for the error code for more details". However, I believe Rust shows very detailed, multi-line error and warning messages, and customers seem to love it. So, maybe we should be moving towards that as well. 🤷‍♂️

Comment (smitpatel) [reply src/Microsoft.NuGet.Build.Tasks/Strings.resx:150]: Missing period in the message in the suggestion.

Comment (JanKrivanek) [reply src/Microsoft.NuGet.Build.Tasks/Strings.resx:150]: I like the suggestion! - it makes it cleaner.

MSB or aka.ms link would be ideal - but there is no detailed info mantioning this (AFAIK) - so I just wanted something that quickly bridges the gap.
When I saw the original error I was confused and after investigating details I was even more confused. After reading the reason for this from @rainersigwald it immediately made sense - and I wished that info was in the message, as it would save me (plus others looking on this customer case) couple hours of head scratching.

**Comment by rainersigwald** [comment]
I would like this error to have an `NUnnnn` code with a corresponding details error page, but it doesn't look like any of the errors in this repo have codes, and they all get reported via a catch-all anyway

https://github.com/dotnet/NuGet.BuildTasks/blob/59a160596088e8e8e7256ab3290a6917a4f8d38e/src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs#L203-L207

It might even be an improvement to have a _single_ searchable/F1-able code for _all_ `ResolveNuGetPackageAssets` errors, with a help page that helps disambiguate (since I assume it's nontrivial to get a bunch of unique codes).

If even that is hard, I support either this or appending only a help link that points to a doc page with this info.

**Review by zivkan — APPROVED** [review]

**Comment by JanKrivanek** [comment]
Closing to rekick the CI

**Comment by JanKrivanek** [comment]
@tmeschter - can you advice how to kick the CI for the PRs in this repo? It seem to be in `Waiting for status to be reported` state for all open PRs

File: src/Microsoft.NuGet.Build.Tasks/Strings.resx [hunk 2]
```diff
@@ -147,7 +147,7 @@
     <comment>0 package id</comment>
   </data>
   <data name="MissingRuntimeIdentifierInProjectFile" xml:space="preserve">
-    <value>Your project file doesn't list '{0}' as a "RuntimeIdentifier". You should add '{1}' to the "RuntimeIdentifiers" property in your project file and then re-run NuGet restore.</value>
+    <value>Your project file doesn't list '{0}' as a "RuntimeIdentifier". You should add '{1}' to the "RuntimeIdentifiers" property in your project file and then re-run NuGet restore. RuntimeIdentifiers should always be the complete list of RIDs supported by the project, regardless of the current configuration, platform or target. A single unconditioned RuntimeIdentifiers property is recommended.</value>
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Microsoft.NuGet.Build.Tasks/Strings.resx:150]: ¯\_(ツ)_/¯ 

I was taking inspiration from MSBuild terminology: https://github.com/dotnet/msbuild/blob/563c7b091b63874dd7e343dc8742690174fe165d/src/Build/Evaluation/IEvaluatorData.cs#L142

But as a non-native speaker I reserve a right to be wrong :-)

[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/Microsoft.NuGet.Build.Tasks/Strings.resx:150]: > But as a non-native speaker I reserve a right to be wrong

Some of the historical authors of MSBuild were worse than that: they were British! /s

Here I would rephrase to avoid:
```suggestion
    <value>Your project file doesn't list '{0}' as a "RuntimeIdentifier". You should add '{1}' to the "RuntimeIdentifiers" property in your project file and then re-run NuGet restore. RuntimeIdentifiers should always be the complete list of RIDs supported by the project, regardless of the current configuration, platform or target. A single RuntimeIdentifiers property with no condition is recommended.</value>
```

**Comment by JanKrivanek** [comment]
Feel free to reopen if there is any interest in the change
